### PR TITLE
feat(tags): add support for tagging schemas

### PR DESCRIPTION
**Pull request by thalesmg**
[EMQX-8330](https://emqx.atlassian.net/browse/EMQX-8330)

By adding tags, it'll be (hopefully) easier to manipulate and split a big schema that has been dumped to JSON based on those.

### CODE REVIEW

File: src/hocon_schema_md.erl [hunk 1]
```diff
@@ -129,6 +132,8 @@ fmt_mapping(Str) -> ["`", Str, "`"].
 
 fmt_type(T) -> hocon_md:code(do_type(T)).
 
+do_type(#{kind := <<Kind/binary>>} = Type) ->
```
Comment (zmstone) [inline src/hocon_schema_md.erl:135]: not quite following why is this needed.

Comment (thalesmg) [reply src/hocon_schema_md.erl:135]: When one reads the schema from an existing JSON file, the fields and values are binaries, so this was blowing up.

It didn't blow up before the refactoring because the schema was processed in-memory, without being written first to disk.  So the key/values stayed as atoms.

(at least that's what I remember, it has been some weeks).

**Review by zmstone — APPROVED** [review]
